src/test/java/core/basesyntax/UserValidationTest.java

```diff
@@ -0,0 +1,161 @@
+        String passwordByPattern2 = "4141&JavaCode";
+        User user = createDefaultUser();
+        user.setPassword(passwordByPattern1);
+        userValidator.validateUser(user);
+        user.setPassword(passwordByPattern2);
+        userValidator.validateUser(user);
+    }
+
+    @Test
+    void validateUser_invalidPassword_NotOk() {
+        List<String> incorrectPasswords = List.of("",
+                "123",
+                "Hello64",
+                "!nowaYaaaaa",
+                "HelloMates!!!",
+                "Yel6low12",
+                "123!#$%^111",
+                "VeryLongPasswordWithA1otofS!ignes"
+        );
+        User user = createDefaultUser();
+        for (String incorrectPassword : incorrectPasswords) {
+            Assertions.assertThrows(RegistrationException.class, () -> {
+                user.setPassword(incorrectPassword);
+                userValidator.validateUser(user);
+            });
+        }
+    }
+
+    @Test
+    void validateUser_negativeAge_NotOk() {
+        User user = createDefaultUser();
+        user.setAge(-2);
+        Assertions.assertThrows(RegistrationException.class, () ->
+                userValidator.validateUser(user));
+    }
+
+    @Test
+    void register_nullableUser_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void register_userWithNullableId_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(new User(null,
+                        DEFAULT_LOGIN,
+                        DEFAULT_VALID_PASSWORD,
+                        DEFAULT_AGE)));
+    }
+
+    @Test
+    void register_userWithNullableLogin_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(new User(DEFAULT_ID,
+                        null,
+                        DEFAULT_VALID_PASSWORD,
+                        DEFAULT_AGE)));
+    }
+
+    @Test
+    void register_userWithNullablePassword_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(new User(DEFAULT_ID,
+                        DEFAULT_LOGIN,
+                        null,
+                        DEFAULT_AGE)));
+    }
+
+    @Test
+    void register_userWithNullableAge_NotOk() {
+        Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(new User(DEFAULT_ID,
+                        DEFAULT_LOGIN,
+                        DEFAULT_VALID_PASSWORD,
+                        null)));
+    }
+
+    @Test
+    void register_UserWithSameLogin_NotOk() throws RegistrationException {
+        User user = new User(1L, "SameLogin123", "HardPassword123!", 18);
+        User newUser = new User(2L, "SameLogin123", "SuperPassword$54", 22);
+
+        registrationService.register(user);
```

Comment:
checklist issue, we can't rely on register to test register, add directly via dao